EasyFinance.Application/Features/ExpenseService/ExpenseService.cs

```diff
@@ -19,76 +23,129 @@ public ExpenseService(IUnitOfWork unitOfWork)
+                return AppResponse<IEnumerable<ExpenseResponseDTO>>.Error(code: nameof(from), description: ValidationMessages.InvalidDate);
+
+            var category = await unitOfWork.CategoryRepository
+                .NoTrackable()
                 .Include(p => p.Expenses.Where(e => e.Date >= from && e.Date < to))
-                    .ThenInclude(e => e.Items.Where(i => i.Date >= from && i.Date < to).OrderBy(item => item.Date))
-                .FirstOrDefaultAsync(p => p.Id == categoryId)).Expenses;
+                .ThenInclude(e => e.Items.Where(i => i.Date >= from && i.Date < to)
+                .OrderBy(item => item.Date))
+                .Skip(paging.PageNumber * paging.PageSize).Take(paging.PageSize)
+                .FirstOrDefaultAsync(p => p.Id == categoryId);
+
+            if (category == null)
+                return AppResponse<IEnumerable<ExpenseResponseDTO>>.Error(code: nameof(categoryId), description: ValidationMessages.CategoryNotFound);
+
+            var expenses = category.Expenses;
+
+            return AppResponse<IEnumerable<ExpenseResponseDTO>>.Success(expenses.ToDTO());
         }
 
-        public async Task<Expense> GetByIdAsync(Guid expenseId)
+        public async Task<AppResponse<ExpenseResponseDTO>> GetByIdAsync(Guid expenseId)
         {
-            return await this.unitOfWork.ExpenseRepository.Trackable()
+            var expense = await unitOfWork.ExpenseRepository.Trackable()
                 .Include(e => e.Items.OrderBy(item => item.Date))
-                    .ThenInclude(e => e.CreatedBy)
+                .ThenInclude(e => e.CreatedBy)
                 .Include(e => e.Attachments)
                 .Include(e => e.CreatedBy)
                 .FirstOrDefaultAsync(p => p.Id == expenseId);
+
+            if (expense == null)
+                return AppResponse<ExpenseResponseDTO>.Error(code: nameof(expense), description: string.Format(ValidationMessages.ExpenseNotFound, nameof(expense)));
+
+            return AppResponse<ExpenseResponseDTO>.Success(expense.ToDTO());
         }
 
-        public async Task<Expense> CreateAsync(User user, Guid categoryId, Expense expense)
+        public async Task<AppResponse<ExpenseResponseDTO>> CreateAsync(User user, Guid categoryId, Expense expense)
         {
             if (expense == default)
-                throw new ArgumentNullException(nameof(expense), string.Format(ValidationMessages.PropertyCantBeNullOrEmpty, nameof(expense)));
+                AppResponse<ExpenseResponseDTO>.Error(code: nameof(expense), description: string.Format(ValidationMessages.PropertyCantBeNullOrEmpty, nameof(expense)));
 
             if (user == default)
-                throw new ArgumentNullException(nameof(user), string.Format(ValidationMessages.PropertyCantBeNullOrEmpty, nameof(user)));
+                AppResponse<ExpenseResponseDTO>.Error(code: nameof(user), description: string.Format(ValidationMessages.PropertyCantBeNullOrEmpty, nameof(user)));
```

Comment:
```suggestion
                return AppResponse<ExpenseResponseDTO>.Error(code: nameof(user), description: string.Format(ValidationMessages.PropertyCantBeNullOrEmpty, nameof(user)));
```
I suggest return here or the code will just run and ignore the error.